dask_kubernetes/tests/test_core.py

```diff
@@ -124,13 +128,15 @@ def test_ipython_display(cluster):
         sleep(0.5)
 
 
-def test_dask_worker_name_env_variable(pod_spec, loop, ns):
+@pytest.mark.asyncio
+async def test_dask_worker_name_env_variable(pod_spec, loop, ns):
```

Comment:
You have `async def` here, but `with KubeCluster` below.  Mixing async and non-async code will probably result in unpleasant hangs.